### ISSUE TITLE
Exception in DataAdapter.Fill when Nullable type is present in query results

### ISSUE THREAD

**Issue by mbtolou** [DarkWanderer/ClickHouse.Client#19]
Exception Nullable Type Generate in sum query.

i cannot simulate on simple query.
i test with `DataReader` and `DataAdaptor.Fill`.

same result generate .

**Comment by mbtolou** [comment]
SELECT toHour(now()) AB,  toNullable(toFloat64(SUM(a)/1.434545)) AA
            from(
                SELECT  number as a from numbers(10000)
            )

**Comment by mbtolou** [comment]
SELECT toHour(now()) AB,  toNullable(toFloat64(SUM(a)/1.434545)) AA
            from(
                SELECT  number as a from numbers(10000)
            )


same result with toInt32,toFloat32,toFloat64

**Comment by mbtolou** [comment]
var dataReader = (ClickHouseDataReader)command.ExecuteReader();

                var dataTable = new DataTable();

                // dataTable.Load(dataReader);


                for (var i = 0; i < dataReader.FieldCount; i++)
                {
                    Type tr = dataReader.GetFieldType(i);
                    if(Nullable.GetUnderlyingType(tr) != null)
                        tr = Nullable.GetUnderlyingType(tr);

                    dataTable.Columns.Add
                    (
                        new DataColumn
                        {
                            ColumnName = dataReader.GetName(i),
                            DataType =  tr,
                            AllowDBNull = true
                        }
                    );
                }

                // dataTable.Load(dataReader);

                while (dataReader.Read())
                {
                    var row = dataTable.NewRow();
                    dataReader.GetValues(row.ItemArray);
                    dataTable.Rows.Add(row);
                }

**Comment by mbtolou** [comment]
i am define 3 Test in my code

TestNullableWithoutAdaptorFillAndDataTableLoad work correct but another test failed
```csharp

[Test]
        public void TestNullableWithoutAdaptorFillAndDataTableLoad()
        {

            using var command = connection.CreateCommand();

            command.CommandText =
            @"SELECT toHour(now()) AB,  toNullable(toFloat64(SUM(a)/1.434545)) AA,  toNullable(toFloat64(SUM(a)/1.4345445)) A1,  toNullable(toFloat64(SUM(a)/1.43234545)) A2
            from(
                SELECT  number as a from numbers(10000)
            )";

            var dataReader = (ClickHouseDataReader)command.ExecuteReader();

            var dataTable = new DataTable();


            for (var i = 0; i < dataReader.FieldCount; i++)
            {
                Type tr = dataReader.GetFieldType(i);
                if (Nullable.GetUnderlyingType(tr) != null)
                    tr = Nullable.GetUnderlyingType(tr);

                dataTable.Columns.Add
                (
                    new DataColumn
                    {
                        ColumnName = dataReader.GetName(i),
                        DataType = tr,
                        AllowDBNull = true
                    }
                );
            }

            while (dataReader.Read())
            {
                var row = dataTable.NewRow();
                for (int i = 0; i < dataReader.FieldCount; i++)
                {
                    row[i] = dataReader[i];
                }
                dataTable.Rows.Add(row);
            }
            Assert.AreEqual(1, dataTable.Rows.Count);

        }


        [Test]
        public void TestNullableWithDataTableLoad()
        {

            using var command = connection.CreateCommand();

            command.CommandText =
            @"SELECT toHour(now()) AB,  toNullable(toFloat64(SUM(a)/1.434545)) AA,  toNullable(toFloat64(SUM(a)/1.4345445)) A1,  toNullable(toFloat64(SUM(a)/1.43234545)) A2
            from(
                SELECT  number as a from numbers(10000)
            )";

            var dataReader = (ClickHouseDataReader)command.ExecuteReader();

            var dataTable = new DataTable();


            dataTable.Load(dataReader);

            Assert.AreEqual(1, dataTable.Rows.Count);

        }

        [Test]
        public void TestNullableWithFill()
        {


            using var adaptor = new ClickHouseDataAdapter();
            using var command = connection.CreateCommand();

            command.CommandText =
            @"SELECT toHour(now()) AB,  toNullable(toFloat64(SUM(a)/1.434545)) AA,  toNullable(toFloat64(SUM(a)/1.4345445)) A1,  toNullable(toFloat64(SUM(a)/1.43234545)) A2
            from(
                SELECT  number as a from numbers(10000)
            )";

            adaptor.SelectCommand = command;

            var dataTable = new DataTable();

            adaptor.Fill(dataTable);

            Assert.AreEqual(1, dataTable.Rows.Count);

            // dataTable.Load(dataReader);



        }
```

**Comment by DarkWanderer** [comment]
Hi,

Thanks for thorough report with repro case. It seems DataSet is not equipped to handle `Nullable` values:
```System.NotSupportedException : DataSet does not support System.Nullable<>.```
I'll see if I can handle this without breaking the type system. As a workaround meanwhile, I suggest using functions like `toFloat64OrZero` or `assumeNotNull` to make sure nullable types don't get to the DataSet

**Comment by mbtolou** [comment]
I need to return null result in my query.

**Comment by DarkWanderer** [comment]
Fixed in 6a59928cd667fddd05aa8ecac3fb0120c276ba92, to be released soon™️

**Comment by DarkWanderer** [comment]
Fixed in [1.1.0.197](https://github.com/DarkWanderer/ClickHouse.Client/releases/tag/1.1.0.197)
